src/Interfaces.ts

```diff
@@ -0,0 +1,60 @@
+import {
+    GraphQLSchema,
+    GraphQLFieldDefinition,
+    GraphQLResolveInfo,
+    GraphQLResult,
+    GraphQLType,
+} from 'graphql';
+
+/* TODO: Add documentation */
+
+export interface IResolverValidationOptions {
+    requireResolversForArgs?: boolean;
+    requireResolversForNonScalar?: boolean;
+    requireResolversForAllFields?: boolean;
+}
+
+export type ITypedef = (() => ITypedef[]) | string;
+export type ITypeDefinitions = ITypedef | ITypedef[];
+export type IResolveTypeFn = (data: any, context: any, info: GraphQLResolveInfo) => any;
+export type IResolveFn = (rootObject?: any, args?: { [key: string]: any }, context?: any, info?: GraphQLResolveInfo) => any;
+export type IResolverObject = { [key: string]: IResolveFn | {resolve?: IResolveFn, __resolveType?: IResolveTypeFn}};
```

Comment:
I think we also have __isTypeOf and not just __resolveType. We're also allowing people to override arbitrary attributes of the type just by prefixing them with __. If we don't want to keep that, then I think we should get rid of __ entriely and just call them resolve, resolveType and isTypeOf.